tools/influx_si/influx_si.xml

```diff
@@ -1,232 +1,74 @@
-            #end if
-            #if $opt.excl_outliers:
-             --excl_outliers '$opt.excl_outliers'
-            #end if
-            $opt.nocalc
-            $opt.addnoise
-            $opt.TIMEIT
-            #if $opt.tblimit:
-             --tblimit='$opt.tblimit'
+<tool id="influx_si" name="influx_si" version="@TOOL_VERSION@+galaxy@VERSION_SUFFIX@" profile="21.09">
+    <description>
+        Estimate metabolic fluxes and concentrations by fitting simulated labeling in metabolites to NMR/MS measurements
+    </description>
+    <creator>
+        <person givenName="Serguei" familyName="Sokol" email="sokol@insa-toulouse.fr" url="https://www.toulouse-biotechnology-institute.fr/en/plateformes-plateaux/cellule-mathematiques/" />
+        <organization name="INRAE" url="https://www.inrae.fr/" />
+        <organization name="TBI" url="https://www.toulouse-biotechnology-institute.fr/" />
+        <organization name="Mathematics Cell" url="https://www.toulouse-biotechnology-institute.fr/en/plateformes-plateaux/cellule-mathematiques/" />
+        <organization name="MetaToul-FluxoMet" url="https://www.toulouse-biotechnology-institute.fr/en/plateformes-plateaux/metatoul/" />
+        <organization name="MetaboHub2" url="https://www.metabohub.fr/" />
+    </creator>
+    <macros>
+        <import>macros.xml</import>
+    </macros>
+    <requirements>
+        <requirement type="package" version="@TOOL_VERSION@">influx_si</requirement>
+    </requirements>
+    <command detect_errors="exit_code"><![CDATA[
+        mkdir outdir;
+        #for $inp in $input_main:
+            unzip $inp;
+        #end for
+               
+        ret=0;
+        influx_$si.s_i
+            @COMMAND_OPT@
+            #if $si.s_i == 'i' and $si.time_order != 'None':
+              --time_order='$si.time_order'
             #end if
+            #for $inp in $input_main:
+                #set base = $inp.name[:-4]
+                --prefix "$base"
+            #end for
+            @MTF_OPT@
+            > 'outdir/influx_${si.s_i}.stdout.txt' 2>'outdir/influx_${si.s_i}.stderr.txt' ||
+            ret=\${PIPESTATUS[0]};
+
 
-         #if $si.s_i == 'i' and $si.time_order:
-           --time_order='$si.time_order'
-         #end if
-         '${"' '".join([str($in_m.element_identifier) for $in_m in $input_main])}'
-         > 'influx_${si.s_i}.stdout.txt' 2>'influx_${si.s_i}.stderr.txt'; res="\$?" &&
-       
-       for f in *.ftbl *.log *.err *.R *.kvh *.attrs; do mv -f "\$f" "\$f".txt || true; done &&
-       
-       for f in *.pdf; do [ -e "\$f" ] && mv -f "\$f" "\$f".pdf || true; done &&
-       
-       for f in *.RData; do [ -e "\$f" ] && mv -f "\$f" "\$f".RData || true; done &&
-       [ "\$res" == "0" ]
+        #for $inp in $input_main:
+            #set base = $inp.name[:-4]
+            for f in "$base"_res/*.{err,log}; do mv -f "\$f" "\$f".txt || true; done;
+            for f in "$base"_res/*.{sim,stat}; do mv -f "\$f" "\$f".tsv || true; done;
```

Comment:
While we are here, at discovery rules, could you give me a hint about a discovery rule for a "high throughput" context?
Influx_si can be used to process the results of dozens or even hundreds experimental results which can share common files, e.g. network configuration but also to have individual files like measurements of various kinds, calculation options and so on. In the end each experiment will have its own directory result, say exp1/ exp2/ etc. The names "expN" are runtime known and dependent on user's requests. So is there a way to discover multiple collections (one per experiment) while their names and number are known only at runtime?

Reply:
The main question is if there is a way to process the experimental results independently, i.e. in separate independent jobs. 

- If yes, then this is ideal for high throughput. On large HPC infrastructure you will get loads of parallel jobs and Galaxy will use the input names (which are ideally in a collection) as output names. The idea is that you have a tool accepting a single dataset and you input a collection. Then Galaxy will automatically create an job for each collection element and it will create an output collection with the same element identifiers as the input collection.
- If no, then it's not true high throughput since you will be limited by the number of cores of one compute node. For the creation of the outputs nested collections would be the way to go.

Reply:
I have managed to write a discovery rule for nested lists but I have lost an automatic file type recognition. Now, if I click on .pdf file, it is automatically downloaded with a cryptic name instead of being viewed as before. Any hint about how to restore file type discovering in this new context?